library/std/src/time.rs

```diff
@@ -458,11 +458,12 @@ impl SystemTime {
     /// # Examples
     ///
     /// ```
-    /// use std::time::SystemTime;
+    /// use std::time::{Duration, SystemTime};
     ///
     /// let sys_time = SystemTime::now();
-    /// let difference = sys_time.duration_since(sys_time)
-    ///                          .expect("Clock may have gone backwards");
+    /// let new_sys_time = sys_time.checked_add(Duration::new(1, 0)).unwrap();
+    /// let difference = new_sys_time.duration_since(sys_time)
+    ///     .expect("Clock may have gone backwards");
```

Comment:
Hmm, I know @pickfire suggested `checked_add` but I don't think it really gets the point of the example across - this will never be an `Err` because the `checked_add` will mean it will always be later than before.

I think I'd prefer `no_run` and having the sleep from before (plus add `use std::thread` at the top):
```suggestion
    /// thread::sleep(Duration::from_secs(1));
    /// let new_sys_time = SystemTime::now();
    /// let difference = new_sys_time.duration_since(sys_time)
    ///     .expect("Clock may have gone backwards");
```

Reply:
@jyn514 I know the previous example is better but it cannot be run with `no_run`, if any I would prefer to see a mock instead of both of the solutions we have right now. It can both tell the `sleep` as well as teaching the readers how to mock time.

Reply:
Right, but I think having `no_run` is better than having an example that doesn't make sense.
Another alternative is to remove the `sleep` altogether, but then it _really_ needs to be no_run because otherwise it will sometimes fail.

Reply:
> Right, but I think having no_run is better than having an example that doesn't make sense.

My suggestion for this is to add a comment that saying it is used to replace `sleep`.

>  Another alternative is to remove the sleep altogether, but then it really needs to be no_run because otherwise it will sometimes fail.

If we do that then this patch does not do anything at all.

Reply:
No, the previous code used the _same_ duration for the comparison, which would never fail because it would never go backwards. This now uses a _new_ duration, which might fail because system time is not monotonic.

Reply:
> This now uses a new duration, which might fail because system time is not monotonic.

Oh, I forgot about this. Yeah, this is why time may go backwards.

Reply:
Glad we're in agreement 😆 

@abrausch do you mind updating this to use `no_run` to make it more clear why this is a potential footgun?
```suggestion
    /// let new_sys_time = SystemTime::now();
    /// let difference = new_sys_time.duration_since(sys_time)
    ///     .expect("Clock may have gone backwards");
```